src/Grpc.Net.Client/Internal/GrpcCallSerializationContext.cs

```diff
@@ -208,7 +210,7 @@ private IBufferWriter<byte> ResolveBufferWriter()
                 // Initialize buffer writer with exact length if available.
                 // ArrayBufferWriter doesn't allow zero initial length.
                 _bufferWriter = _payloadLength > 0
-                    ? new ArrayBufferWriter<byte>(_payloadLength.GetValueOrDefault())
+                    ? new ArrayBufferWriter<byte>(_payloadLength.Value)
                     : new ArrayBufferWriter<byte>();
```

Comment:
Here the JIT can't prove that `_payloadLength` has a non-null value, as the check for `DirectSerializationSupported` could return `false` if `__compressionProvider != null` too.
I'd write this as
```c#
            var payloadLength = _payloadLength.GetValueOrDefault();
            
            // Initialize buffer writer with exact length if available.
            // ArrayBufferWriter doesn't allow zero initial length.
            _bufferWriter = payloadLength > 0
                ? new ArrayBufferWriter<byte>(payloadLength)
                : new ArrayBufferWriter<byte>();
```

Both suggestions lead to the seconds method in the above sharplab, no exception-case here.

Reply:
if `_payloadLength` is greater than zero then it can't be null, right?

Reply:
Yes.

If it's null, then the default value is `0`, so the ternary does the right thing.